test/NuGet.Clients.FuncTests/NuGet.CommandLine.FuncTest/Commands/RestoreCommandTests.cs

```diff
@@ -146,6 +150,285 @@ public async Task Restore_LegacyPackageReference_WithNuGetLockFilePath()
             }
         }
 
+        [Fact]
+        public async Task Restore_LegacyPackageReference_WithNuGetLockFileArgument()
+        {
+            // Arrange
+            using (var pathContext = new SimpleTestPathContext())
+            {
+                // Set up solution, project, and packages
+                var solution = new SimpleTestSolutionContext(pathContext.SolutionRoot);
+
+                var net461 = NuGetFramework.Parse("net461");
+
+                var projectA = SimpleTestProjectContext.CreateLegacyPackageReference(
+                    "a",
+                    pathContext.SolutionRoot,
+                    net461);
+
+                var packageX = new SimpleTestPackageContext()
+                {
+                    Id = "x",
+                    Version = "1.0.0"
+                };
+                packageX.Files.Clear();
+                packageX.AddFile("lib/net461/a.dll");
+
+                projectA.AddPackageToAllFrameworks(packageX);
+                solution.Projects.Add(projectA);
+                solution.Create(pathContext.SolutionRoot);
+
+                await SimpleTestPackageUtility.CreateFolderFeedV3Async(
+                    pathContext.PackageSource,
+                    PackageSaveMode.Defaultv3,
+                    packageX);
+
+                // Act
+                var result = RunRestore(pathContext, _successExitCode, "-UseLockFile", "true");
+
+                // Assert
+                Assert.True(File.Exists(projectA.NuGetLockFileOutputPath));
+
+                var lockFile = PackagesLockFileFormat.Read(projectA.NuGetLockFileOutputPath);
+                Assert.Equal(4, lockFile.Targets.Count);
```

Comment:
One target is validated below.  What are the other 3?

In other words, imagine this fails in the future.  How can I know what the expected targets are?  Is it even important to have this assertion in this test?

Reply:
The other 3 are the same TFM with 3 different RIDs. I'm surprised that the RID versions don't have the same dependency that the RIDless target has, but I don't understand the asset file well enough.

Reply:
I forgot to mention that the test is almost exactly the same as the test above, only using the nuget.exe argument instead of a MSBuild property.

Reply:
I think that some test should verify the other 3 targets and if it's not this test then this test can simply assert:

    Assert.True(lockFile.Targets.Count >= 1); // 1 RID-less graph + 0..n RID graphs

Same for the other tests.  Just because some test you copied has a poor assertion doesn't mean we should carry that forward in future tests.